Automattic/vip-go-mu-plugins
# Replace ElasticPress network layer with vip_safe_wp_remote_get()

## Description

Switch out the default WP HTTP API for Elasticsearch requests with `vip_wp_safe_remote_get()` for added safety (timeout and limited global failure counts).

Depends on #1395

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. On a local env with Elasticsearch...
1. Switch to this branch
1. Perform some searches regularly
1. Reindex everything
1. Reindex single item
1. Verify above all behaves as expected
1. Stop `elasticsearch` container
1. Repeat above, but verify that requests are erroring with the expected error from our interception layer (`There was an error connecting to the upstream Elasticsearch server`). This can be viewed in the ElasticPress Debug Bar panel
